From 52bdfb0fca9ccf2597f528d54a73e986e760cd65 Mon Sep 17 00:00:00 2001
From: Shadi Ammouri <shadi@marvell.com>
Date: Thu, 17 Dec 2015 14:59:01 +0200
Subject: [PATCH 23/66] kernel: Fix bug when using defconfig as the kernel's
 defconfig file

	Patch provided by Thomas Petazzoni <thomas.petazzoni@free-electrons.com>.

Change-Id: Ia27be2a94c617c77e8c72df483d6c83a88ec01c3
Signed-off-by: Shadi Ammouri <shadi@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/26092
---
 linux/Config.in | 3 +++
 linux/linux.mk  | 2 ++
 2 files changed, 5 insertions(+)

diff --git a/linux/Config.in b/linux/Config.in
index 12b417a..3093ec2 100644
--- a/linux/Config.in
+++ b/linux/Config.in
@@ -142,6 +142,9 @@ choice
 config BR2_LINUX_KERNEL_USE_DEFCONFIG
 	bool "Using an in-tree defconfig file"
 
+config BR2_LINUX_KERNEL_USE_DEFAULT_DEFCONFIG
+	bool "Using the architecture default defconfig"
+
 config BR2_LINUX_KERNEL_USE_CUSTOM_CONFIG
 	bool "Using a custom (def)config file"
 
diff --git a/linux/linux.mk b/linux/linux.mk
index 9507837..6919e86 100644
--- a/linux/linux.mk
+++ b/linux/linux.mk
@@ -169,6 +169,8 @@ LINUX_POST_PATCH_HOOKS += LINUX_APPLY_LOCAL_PATCHES
 
 ifeq ($(BR2_LINUX_KERNEL_USE_DEFCONFIG),y)
 KERNEL_SOURCE_CONFIG = $(KERNEL_ARCH_PATH)/configs/$(call qstrip,$(BR2_LINUX_KERNEL_DEFCONFIG))_defconfig
+else ifeq ($(BR2_LINUX_KERNEL_USE_DEFAULT_DEFCONFIG),y)
+KERNEL_SOURCE_CONFIG = $(KERNEL_ARCH_PATH)/configs/defconfig
 else ifeq ($(BR2_LINUX_KERNEL_USE_CUSTOM_CONFIG),y)
 KERNEL_SOURCE_CONFIG = $(call qstrip,$(BR2_LINUX_KERNEL_CUSTOM_CONFIG_FILE))
 endif
-- 
1.9.1

